pennylane/transforms/control.py

```diff
@@ -0,0 +1,191 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the control transform.
+"""
+
+
+from functools import wraps
+from pennylane.tape import QuantumTape
+from pennylane.operation import Operation, AnyWires
+from pennylane.wires import Wires
+from pennylane.transforms.registrations import register_control_transform, CONTROL_MAPS
+from pennylane.transforms.adjoint import adjoint
+
+
+def expand_with_control(tape, control_wire):
+    """Expand a tape to include a control wire on all queued operations.
+
+    Args:
+        tape (.QuantumTape): quantum tape to be controlled
+        control_wire (int): a single wire to use as the control wire
+
+    Returns:
+        .QuantumTape: A new QuantumTape with the controlled operations.
+    """
+    with QuantumTape(do_queue=False) as new_tape:
+        for op in tape.operations:
+            if op.__class__ in CONTROL_MAPS:
+                # Create the controlled version of the operation
+                # and add that the to the tape context.
+                CONTROL_MAPS[op.__class__](op, control_wire)
+            else:
+                raise NotImplementedError(
+                    f"Control transform of operation {type(op)} is not registered."
+                    "You can define the control transform by using "
+                    "qml.register_control_transform. See documentation for more details."
+                )
+    return new_tape
+
+
+class ControlledOperation(Operation):
+    """A Controlled Operation.
+
+    Unless you are a Pennylane plugin developer, **you should NOT directly use this class**,
+    instead, use the :func:`qml.ctrl <.ctrl>` function.
+
+    The ``ControlledOperation`` class is a container class that defines a set of operations that
+    should by applied relative to a single control wire or a list of control wires.
+
+    Certain simulators and quantum computers can take advantage of the controlled gate sparsity,
+    while other devices must rely on the op-by-op decomposition defined by the ``op.expand``
+    method.
+
+    Args:
+        tape: A QuantumTape. This tape defines the unitary that should be applied relative
+            to the control wires.
+        control_wires: A wire or set of wires.
+    """
+
+    par_domain = "A"
+    num_wires = AnyWires
+    num_params = property(lambda self: self.subtape.num_params)
+
+    def __init__(self, tape, control_wires, do_queue=True):
+        self.subtape = tape
+        """(QuantumTape): The tape that defines the underlying operation."""
+
+        self.control_wires = Wires(control_wires)
+        """(Wires): The control wires."""

```

Comment:
```suggestion
        """QuantumTape: The tape that defines the underlying operation."""

        self.control_wires = Wires(control_wires)
        """Wires: The control wires."""
```
(for attribute docstrings, the parenthesis are not required)